contributed_definitions/nyaml/NXbeam.yaml

```diff
@@ -74,133 +88,155 @@ category: base
         this is an array of length m with the relative weights in incident_wavelength_weights as a 2D array.
       * In the case of a polychromatic beam that varies shot-to-shot and where the channels also vary,
         this is a 2D array of dimensions nP by m (slow to fast) with the relative weights in incident_wavelength_weights as a 2D array.
-
+      
       Note, variants are a good way to represent several of these use cases in a single dataset, e.g. if a calibrated,
       single-value wavelength value is available along with the original spectrum from which it was calibrated.
     dimensions:
       rank: 1
       dim: [[1, i]]
   incident_wavelength_spread(NX_FLOAT):
     unit: NX_WAVELENGTH
-    doc: "Wavelength spread FWHM on entering component"
+    doc: |       
+      Wavelength spread FWHM on entering component
     dimensions:
       rank: 1
       dim: [[1, i]]
   incident_beam_divergence(NX_FLOAT):
     unit: NX_ANGLE
-    doc: "Divergence of beam entering this component"
+    doc: |       
+      Divergence of beam entering this component
     dimensions:
       rank: 2
       dim: [[1, 2], [2, j]]
   extent(NX_FLOAT):
     unit: NX_LENGTH
-    doc: "Size of the beam entering this component"
+    doc: |       
+      Size of the beam entering this component
     dimensions:
       rank: 2
       dim: [[1, 2], [2, j]]
   final_wavelength(NX_FLOAT):
     unit: NX_WAVELENGTH
-    doc: "Wavelength on leaving beamline component"
+    doc: |       
+      Wavelength on leaving beamline component
     dimensions:
       rank: 1
       dim: [[1, i]]
   incident_polarization(NX_FLOAT):
     unit: NX_ANY
-    doc: "Incident polarization as a Stokes vector"
+    doc: |       
+      Incident polarization as a Stokes vector
     dimensions:
       rank: 1
       dim: [[1, 4]]
     \@units:
-      doc: |
+      type: NX_CHAR
```

Comment:
Typically, NX_CHAR should be omitted, because it is set automatically if no type is given.

Reply:
it should be there because the rule is currently that no automation is taken care of by nyaml2nxdl so if the developer has nxchar added it will be added if not not